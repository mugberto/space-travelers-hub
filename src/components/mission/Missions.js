import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import MissionsTable from './MissionsTable';
import { getMissions, joinMission, leaveMission } from '../../redux/mission/mission';

const Missions = () => {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.missions);

  useEffect(() => {
    if (!missions.length) {
      dispatch(getMissions);
    }
  }, []);

  const handleJoining = (id) => dispatch(joinMission(id));
  const handleLeaving = (id) => dispatch(leaveMission(id));

  return (
    <MissionsTable
      missions={missions}
      handleJoining={handleJoining}
      handleLeaving={handleLeaving}
    />
  );
};

export default Missions;
