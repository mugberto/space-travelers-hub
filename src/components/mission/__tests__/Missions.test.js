import React from 'react';
import { render, fireEvent, screen } from '../../../test-utils';
import Missions from '../Missions';

describe('Missions', () => {
  it('renders page correctly', () => {
    const tree = render(<Missions />);
    expect(tree).toMatchSnapshot();
  });

  it('updates the UI with loaded missions', async () => {
    render(<Missions />);
    expect(await screen.findByText('Thaicom')).toBeInTheDocument();
  });

  it('notifies the user of the mission join status', async () => {
    render(<Missions />);
    expect(await screen.findByText('NOT A MEMBER')).toBeInTheDocument();
  });

  it('updates the mission join status when the user clicks on the "Join Mission" button', async () => {
    render(<Missions />);
    expect(await screen.findByText('NOT A MEMBER')).toBeInTheDocument();
    fireEvent.click(screen.getByRole('button', { name: 'Join Mission' }));
    expect(await screen.findByText('Active Member')).toBeInTheDocument();
  });

  it('updates the mission join status when the user clicks on the "Leave Mission" button', async () => {
    render(<Missions />);
    expect(await screen.findByText('NOT A MEMBER')).toBeInTheDocument();
    fireEvent.click(screen.getByRole('button', { name: 'Join Mission' }));
    expect(await screen.findByText('Active Member')).toBeInTheDocument();
    fireEvent.click(screen.getByRole('button', { name: 'Leave Mission' }));
    expect(await screen.findByText('NOT A MEMBER')).toBeInTheDocument();
  });
});
