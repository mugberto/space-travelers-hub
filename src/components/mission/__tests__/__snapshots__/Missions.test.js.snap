// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions renders page correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1fwtis1-MuiPaper-root"
      >
        <div
          class="MuiTableContainer-root css-lgwwp6-MuiTableContainer-root"
        >
          <table
            aria-label="sticky table"
            class="MuiTable-root MuiTable-stickyHeader css-1x6isbw-MuiTable-root"
          >
            <thead
              class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head css-1xzp0f1-MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-40tvjn-MuiTableCell-root"
                  scope="col"
                >
                  Mission
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignLeft MuiTableCell-sizeMedium css-40tvjn-MuiTableCell-root"
                  scope="col"
                >
                  Description
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignCenter MuiTableCell-sizeMedium css-l6zyo1-MuiTableCell-root"
                  scope="col"
                >
                  Status
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignLeft MuiTableCell-sizeMedium css-40tvjn-MuiTableCell-root"
                  scope="col"
                >
                    
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
            />
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1fwtis1-MuiPaper-root"
    >
      <div
        class="MuiTableContainer-root css-lgwwp6-MuiTableContainer-root"
      >
        <table
          aria-label="sticky table"
          class="MuiTable-root MuiTable-stickyHeader css-1x6isbw-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1xzp0f1-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-40tvjn-MuiTableCell-root"
                scope="col"
              >
                Mission
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignLeft MuiTableCell-sizeMedium css-40tvjn-MuiTableCell-root"
                scope="col"
              >
                Description
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignCenter MuiTableCell-sizeMedium css-l6zyo1-MuiTableCell-root"
                scope="col"
              >
                Status
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignLeft MuiTableCell-sizeMedium css-40tvjn-MuiTableCell-root"
                scope="col"
              >
                  
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          />
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
